samples/java-protobuf-eventsourced-counter/src/it/java/com/example/CounterTopicIntegrationTest.java

```diff
@@ -0,0 +1,68 @@
+/* This code was generated by Kalix tooling.
+ * As long as this file exists it will not be re-generated.
+ * You are free to make changes to this file.
+ */
+package com.example;
+
+import com.example.actions.CounterTopicApi;
+import kalix.javasdk.testkit.EventingTestKit;
+import kalix.javasdk.testkit.junit.KalixTestKitResource;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.concurrent.CompletionStage;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.Assert.assertEquals;
+
+// Example of an integration test calling our service via the Kalix proxy
+// Run all test classes ending with "IntegrationTest" using `mvn verify -Pit`
+public class CounterTopicIntegrationTest {
+
+  /**
+   * The test kit starts both the service container and the Kalix proxy.
+   */
+  @ClassRule
+  public static final KalixTestKitResource testKit =
+      new KalixTestKitResource(Main.createKalix());
+
+  /**
+   * Use the generated gRPC client to call the service through the Kalix proxy.
+   */
+  private final CounterService client;
+
+  private final EventingTestKit.Topic commandsTopic;
+  private final EventingTestKit.Topic eventsTopic;
+
+
+  public CounterTopicIntegrationTest() {
+    client = testKit.getGrpcClient(CounterService.class);
+    commandsTopic = testKit.getTopic("counter-commands");
+    eventsTopic = testKit.getTopic("counter-events");
+  }
+
+  private int getCounterValue(String counterId) throws Exception {
+    CounterApi.GetCounter getCounter = CounterApi.GetCounter.newBuilder().setCounterId(counterId).build();
+    return awaitResult(client.getCurrentCounter(getCounter)).getValue();
+  }
+
+  private <T> T awaitResult(CompletionStage<T> stage) throws Exception {
+    return stage.toCompletableFuture().get(5, SECONDS);
+  }
+  @Test
+  public void verifyCounterCommandsAndPublish() throws Exception {
+    var counterId = "test-topic";
+
+    var increaseCmd = CounterApi.IncreaseValue.newBuilder().setCounterId(counterId).setValue(4).build();
+    var decreaseCmd = CounterApi.DecreaseValue.newBuilder().setCounterId(counterId).setValue(1).build();
+    commandsTopic.publish(increaseCmd, counterId);
+    commandsTopic.publish(decreaseCmd, counterId);
+
+    var increasedEvent = eventsTopic.expectOneTyped(CounterTopicApi.Increased.class);
+    var decreasedEvent = eventsTopic.expectOneTyped(CounterTopicApi.Decreased.class);
+    assertEquals(increaseCmd.getValue(), increasedEvent.getPayload().getValue());
+    assertEquals(decreaseCmd.getValue(), decreasedEvent.getPayload().getValue());
+
+    assertEquals(3, getCounterValue(counterId));
```

Comment:
this assertion is not present in the Scala sample, maybe we could remove from here?